src/lsp/cobol_parser/keywords/gen_keywords.ml

```diff
@@ -174,6 +174,20 @@ let emit_intrinsic_functions_list ppf =
   Terminal.iter (emit_custom_intrinsics ppf);
   Fmt.pf ppf "@]@\n]@."
 
+let emit_is_intrinsic ppf =
+  if String.equal cmlyname "grammar.cmly" then
+  let is_intrinsic t =
+    intrinsic (Terminal.attributes t) |> Option.is_some
+  in
+  Fmt.pf ppf "@[<2>let is_intrinsic_token = %s.(function@." tokens_module;
+  Terminal.iter begin fun t ->
+    if is_intrinsic t
+    then Fmt.pf ppf "| %a@." pp_terminal t
+  end;
+  Fmt.pf ppf "| INTRINSIC_FUNC _ -> true@.";
```

Comment:
A bit hardcoded as well. Maybe that case can be taken out, with a refined semantics, renamed like `is_known_intrinsics` (and a need to match `INTRISIC_FUNC` outside of this function).